### PR TITLE
Release Nightscout 0.12.4

### PR DESCRIPTION
**Pull request by PieterGit**
Can all of you please help test current dev and report if it works for you? I would like to release a patch release at the end of august if nothing pops up. Main fix is to be able to use Medtronic GuardianConnect in the US. If there is stuff that really needs to be merged in 0.12.4, please speak up now.

TODO: draft up release notes.

Please report success or issues with the upcoming 0.12.4 release here.

### CODE REVIEW

File: lib/report_plugins/daytoday.js [hunk 1]
```diff
@@ -738,7 +738,6 @@ daytoday.report = function report_daytoday (datastorage, sorteddaystoshow, optio
         var label = ' ' + treatment.carbs + ' g';
         if (treatment.protein) label += ' / ' + treatment.protein + ' g';
         if (treatment.fat) label += ' / ' + treatment.fat + ' g';
-        label += ' (' + client.utils.toFixedMin((treatment.carbs / ic), 2) + 'U)';
```
Comment (sulkaharo) [inline lib/report_plugins/daytoday.js]: Small cleanup missing - with this gone, the ic variable is unused

> cgm-remote-monitor/lib/report_plugins/daytoday.js
> 737:13  error  'ic' is assigned a value but never used  no-unused-vars

**Comment by PetrOndrusek** [comment]
I have noticed some cosmetic issues - on the dev branch there a 3 unit test failing (not ok).

- **not ok 1 reports should produce some html**
  There is a 'Unable to find element for #chartContainer' message in the output and reports did work for me during my testing, so maybe only the test needs corrections. 

- **not ok 2 utils format numbers short**
AssertionError: expected 'NaN' to be '0'
This is a problem of utils.toFixedMin function which is not yet referenced anywhere in the project, so it should be harmless

- **not ok 17 Entries REST api post an entry, query, delete, verify gone**
  Error: expected 200 "OK", got 401 "Unauthorized"
  It looks like only self.env.api_secret is not set for the test in Travis CI environment, on my local dev environment this test succeeds

**Comment by jpcunningh** [comment]
We updated to the dev branch this evening.

So far, everything appears to operating correctly for us using OpenAPS and Lookout.  It seemed to take noticeably longer to load than it used to.

I have several npm tests failing on my local development environment, but the failures appear to be issues with the test, not the operational functionality.

**Comment by sulkaharo** [comment]
Release notes draft:

 Nightscout 0.12.4

* Most Javascript in the site is now transpiled (yes this is highly technical) to be compatible with older devices such as iPads running iOS 9, so devices that haven’t worked correctly with release 0.12 probably work after upgrading to 0.12.4
* Medtronic Guardian Connect Plugin has been updated to be compatible with US users
* Clock views now show the X to close on load, which then hides itself until screen is clicked
* Several bug fixes to the clock views
* Daily reports don’t show estimated insulin values for carbs due to potential confusion over this referring to dosed insulin
* Profile conversion warnings were disabled
* Fixed a plugin setting rendering bug introduced in the 0.12.3
* German and Russian language updates
* Mmol unit conversion fixes
* Several small documentation improvements
* Fixed an issue causing a client crash on first start of a new site

Contributor thanks (in no particular order) to this release go to @Erin879, @GermanBluefox, @PieterGit, @PetrOndrusek, @jpcunningh, @codacy-bot, @bassettb, @ireneusz-ptak, @unsoluble, @herzogmedia, @aviau, @apanasef & @sulkaharo

**Pull request merged**